packages/ui/hooks/useDetails.tsx

```diff
@@ -0,0 +1,22 @@
+import { useBoolean } from '@chakra-ui/react';
+import { useCallback, useEffect, useRef } from 'react';
+
+export function useDetails(): readonly [
```

Comment:
This looks like it will work though I'm not sure I understand why it's necessary. The places where we use isOpen are pretty limited, just showing different text for different open/closed states, which I would be okay with dropping or using a pure css solution like:

```html
<details>
  <summary>
    <span class="whenClosed">Show More</span>
    <span class="whenOpen">Show Fewer</span>
   </summary
   Stuff goes here
</details>
```
```css
details .whenOpen {
   display: none;
}
details[open] .whenOpen {
   display: initial;
}
details[open] .whenClosed {
   display: none;
}
```

For activating the toggle behavior when clicking the outer component, if we don't need to track the open/closed state, we might be able to get away without a ref or useDetails if the click handler is like:

```js
onClick: (e) => {
   e.currentTarget.getElementsByTagName('details')[0]?.toggleProperty('open');
}
```

Reply:
You're right that we don't necessarily need a custom hook for details and chakra wrapper for Summary. However, it's a departure from the pattern of using hooks and Chakra components (instead of native DOM accessors and pure CSS) that we regularly use in this repo. Your solution might be preferable, but it seems more like a conversation of "Should we use native features (potentially  more performant styling without JS) instead of convenience Chakra wrappers?".

When I wrote the useDetails hook, I was trying to mimic Chakra's [useDisclosure](https://chakra-ui.com/docs/hooks/use-disclosure#return-value) hook, for the record.

Reply:
I agree it's a departure and 99% of the time I'd be against it, since it's so counter to the react way of doing things. On the other hand, if we can get away without worrying about the open state of the details element in js, then I'd rather keep it simple.

Reply:
BTW with regard to being more performant vs more convenient, I think we've made the decision to be more convenient a lot of times. However, I'm not convinced that in this case the convenience tradeoff is there, especially if you use your new chakra summary element. I think we should see how this looks without the hook and then decide.

Reply:
👍🏻 

I put the CSS as `sx` in a container component to keep it closer to the html. If you think it's better to put it in a dedicated css file just lmk

https://github.com/iFixit/react-commerce/pull/1604/commits/f9ae8642b031243af7163955453d7340813796ab